getnacelle/sanity-preview-connector
# Scaffold Preview Connector

### WHY are these changes introduced?

Fixes [DRMS-871](https://nacelle.atlassian.net/browse/DRMS-871)

> As a merchant developer, I want to fetch Sanity preview content with the Nacelle SDK, so that my team can preview content before it's published



### WHAT is this pull request doing?

- Enables draft content to be fetched by [`@nacelle/client-js-sdk`](https://www.npmjs.com/package/@nacelle/client-js-sdk):

https://user-images.githubusercontent.com/5732000/107786971-80105d00-6d1c-11eb-8dc8-e890b5ab3f0e.mov

## 

- Adds unit tests
- Adds example code for Nuxt and Vanilla JavaScript projects

### How to Test

1. `npm run test`
2. Create `examples/vanilla/.env` with `SANITY_PROJECT_ID`, `SANITY_DATASET`, and `SANITY_TOKEN` environment variables
3. In `examples/vanilla`, `npm run dev`
4. Navigate to `localhost:3000` and select a content type of interest
5. Create unpublished content in Sanity and click the "refresh" button in the example project to see your unpublished content

### Testing checklist

* [ ] Updated public facing documentation (TODO: [DRMS-895](https://nacelle.atlassian.net/browse/DRMS-895))
